app/shared/layouts/navigation/SegmentIdentify.js

```diff
@@ -6,30 +6,30 @@ import { useSearchParams } from 'next/navigation'
 import { getPersistedUtms } from 'helpers/analyticsHelper'
 import { extractClids } from 'helpers/analyticsHelper'
 import { useEffect } from 'react'
-import { useAnalytics } from '@shared/hooks/useAnalytics'
+import { analytics } from '@shared/utils/analytics'
 
-export default function SegmentIdentify({extraTraits = {}}) {
-  const analytics = useAnalytics()
+export default function SegmentIdentify() {
   const [user] = useUser()
   const searchParams = useSearchParams()
   useEffect(() => {
 
-    if (!analytics || !analytics?.user || analytics.user().id()) return
-    persistUtmsOnce()
+    if (!analytics) return
+    (async () => {
+      const analyticsUser = await analytics.user()
+      if (analyticsUser.id() || analyticsUser.anonymousId()) return // No need to spam identity calls that have no new information
 
-    if (user?.id) {
-      
-      analytics.identify(user.id, {
-        name: `${user.firstName} ${user.lastName}`,
-        email: user.email,
-        phone: user.phone,
-        zip:   user.zip,
-        ...extraTraits,
-        ...getPersistedUtms(), 
-        ...extractClids(searchParams) 
-      })
-    } else {
-      analytics.identify({ ...extraTraits })
-    }
-  }, [user, searchParams, extraTraits, analytics])
+      persistUtmsOnce()
+
+      user?.id 
+      ? analytics.identify(user.id, {
+          name: `${user.firstName} ${user.lastName}`,
+          email: user.email,
+          phone: user.phone,
+          zip:   user.zip,
+          ...getPersistedUtms(), 
+          ...extractClids(searchParams) 
+        })
+      : analytics.identify()
+    })()
```

Comment:
Why the IIFE here?

Seems simple enough define a named method and then just do:


```suggestion
return analytics && await identify(user) 
```

Now that I think of it.. this method should prolly just be in the `helpers/analyticsHelper` anyway..